orangecontrib/text/corpus.py

```diff
@@ -100,6 +100,22 @@ def extend_corpus(self, metadata, Y):
         self.X = self.W = np.zeros((len(self), 0))
         Table._init_ids(self)
 
+    def update_attributes(self, X, dictionary):
+        """
+        Adds X to the corpus and updates its domain accordingly.
+        """
+        self.X = np.hstack((self.X, X))
+
+        dict_key_count = len(sorted(dictionary.keys()))
+        features = [dictionary[key] for key in range(dict_key_count)]
+        attr = [ContinuousVariable.make(f) for f in features]
+        new_domain = Domain(
+                attributes=attr,
```

Comment:
If the user construct two consecutive BoW widgets on purpose than that is what he/she should receive. I do not know why this could be useful, but I still think that it is worse if we delete user-provided features and just replace them with BoW.